addons/product_expiry/models/stock_move_line.py

```diff
@@ -14,10 +14,12 @@ class StockMoveLine(models.Model):
         help='This is the date on which the goods with this Serial Number may'
         ' become dangerous and must not be consumed.')
 
-    @api.depends('product_id', 'picking_type_use_create_lots')
+    @api.depends('product_id', 'picking_type_use_create_lots', 'lot_id.expiration_date')
     def _compute_expiration_date(self):
         for move_line in self:
-            if move_line.picking_type_use_create_lots:
+            if not move_line.expiration_date and move_line.lot_id.expiration_date:
```

Comment:
I wanted to suggest you to update the `move_line.expiration_date` even if it's already set (you create the lot with its expiration date, then re-open the lot form and modify the expiration date) but after trying manually that, it seems the compute acts weirdly in this case...
It will go in it, but the `expiration_date` will not be really updated.
It seems that's because no move line's fields are modified. If you change the `lot_id` or any other fields, the new value for the compute will be saved.

I'll try something else.

Reply:
Nevermind, I tried other solutions but it becomes complicated to have something working in all cases. I guess we can stay like that for now if there is no further complains and eventually improve that in master.